pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java

```diff
@@ -489,24 +489,27 @@ public CompletableFuture<Optional<String>> selectAsync(ServiceUnitId bundle,
 
                     // Filter out brokers that do not meet the rules.
                     List<BrokerFilter> filterPipeline = getBrokerFilterPipeline();
+                    ArrayList<CompletableFuture<Map<String, BrokerLookupData>>> futures =
+                            new ArrayList<>(filterPipeline.size());
                     for (final BrokerFilter filter : filterPipeline) {
-                        try {
-                            filter.filter(availableBrokerCandidates, bundle, context);
-                            // Preserve the filter successes result.
-                            availableBrokers.keySet().retainAll(availableBrokerCandidates.keySet());
-                        } catch (BrokerFilterException e) {
-                            // TODO: We may need to revisit this error case.
-                            log.error("Failed to filter out brokers.", e);
-                            availableBrokerCandidates = new HashMap<>(availableBrokers);
-                        }
-                    }
-                    if (availableBrokerCandidates.isEmpty()) {
-                        return CompletableFuture.completedFuture(Optional.empty());
+                        CompletableFuture<Map<String, BrokerLookupData>> future =
+                                filter.filter(availableBrokerCandidates, bundle, context);
+                        futures.add(future);
```

Comment:
do we need process order for the filter here. the k8s scheduler will apply filter orderly. I wonder if we need it.

Reply:
No, we do no need to guarantee the order.